src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -623,6 +623,11 @@ protected function redirectAfterSave($isNewRecord)
             // to the same URL (it assumes that its content is already current, and doesn't reload)
             return $this->edit($controller->getRequest());
         } else {
+            // We might be able to redirect to open the record in a different view
+            if ($redirectDest = $this->component->getLostRecordRedirection($this->gridField, $controller->getRequest(), $this->record->ID)) {
```

Comment:
Should check that it's not already on that URL to avoid infinite redirection loops. The CMSEditLink might link to a prefiltered view as well where the record doesn't exist either.

Reply:
Yeah I actually had an infinite redirection in my project at first, due to faulty filtering / CMSEditLink logic. I lost the check along the way somewhere. Will recreate.

Reply:
I've added some basic infinite loop direction. Only checks for exact matches of the URL, so could still be thrown off by GET params etc, but it'll do.